src/objects/SkinnedMesh.js

```diff
@@ -182,6 +195,83 @@ SkinnedMesh.prototype = Object.assign( Object.create( Mesh.prototype ), {
 
 		return new this.constructor( this.geometry, this.material, this.skeleton.useVertexTexture ).copy( this );
 
+	},
+
+	toJSON: function ( meta ) {
+
+		var isRootObject = ( meta === undefined || meta === '' );
+
+		if ( isRootObject ) {
+
+			meta = {
+				geometries: {},
+				materials: {},
+				textures: {},
+				images: {},
+				skeletons: {}
+			};
+
+		}
+
+		var data = THREE.Mesh.prototype.toJSON.call( this, meta );
+
+		if ( isRootObject ) {
+
+			data.metadata = {
+				version: 4.4,
+				type: 'SkinnedMesh',
+				generator: 'SkinnedMesh.toJSON'
+			};
+
+
+		}
+
+		var object = data.object;
+
+		object.bindMode = this.bindMode;
+		object.bindMatrix = this.bindMatrix.toArray();
+
+		if ( meta.skeletons[ this.skeleton.uuid ] === undefined ) {
+
+			meta.skeletons[ this.skeleton.uuid ] = this.skeleton.toJSON( meta );
+			object.skeleton = this.skeleton.uuid;
+
+		}
+
+		// copied from Object3D
+		if ( isRootObject ) {
+
+			var geometries = extractFromCache( meta.geometries );
+			var materials = extractFromCache( meta.materials );
+			var textures = extractFromCache( meta.textures );
+			var images = extractFromCache( meta.images );
+			var skeletons = extractFromCache( meta.skeletons );
+
+			if ( geometries.length > 0 ) data.geometries = geometries;
+			if ( materials.length > 0 ) data.materials = materials;
+			if ( textures.length > 0 ) data.textures = textures;
+			if ( images.length > 0 ) data.images = images;
+			if ( skeletons.length > 0 ) data.skeletons = skeletons;
+
+		}
+
+		return data;
+
+		// copied from Object3D
+		function extractFromCache ( cache ) {
+
+			var values = [];
+			for ( var key in cache ) {
+
+				var data = cache[ key ];
+				delete data.metadata;
+				values.push( data );
+
+			}
+			return values;
+
+		}
+
```

Comment:
Wondering if I can remove a lot of duplicated lines with `Object3D.toJSON()`... any ideas?